### PR TITLE
Clean up foodcritic errors.

### LINKED ISSUE #190: Broken build
Failing build: https://travis-ci.org/sous-chefs/rundeck/jobs/421278908

Build failure message:
```
FC113: Resource declares deprecated use_inline_resources: ./providers/plugin.rb:1
FC113: Resource declares deprecated use_inline_resources: ./providers/plugin.rb:21
FC113: Resource declares deprecated use_inline_resources: ./providers/project.rb:1
FC113: Resource declares deprecated use_inline_resources: ./providers/user.rb:1
FC113: Resource declares deprecated use_inline_resources: ./providers/user.rb:21
FC120: Do not set the name property directly on a resource: ./recipes/server_install.rb:68
FC121: Cookbook depends on cookbook made obsolete by Chef 14: ./metadata.rb:1
FC122: Use the build_essential resource instead of the recipe: ./recipes/_connect_rundeck_api_client.rb:3
```

Hopefully this may be low hanging fruit!

### PR DESCRIPTION
**Pull request by mrala**
Resolves #190 

- Remove deprecated use_inline_resources syntax
- Use `build_essential` resource instead of cookbook

- [ ] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable

### CODE REVIEW

File: recipes/_connect_rundeck_api_client.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # rest-client gem is needed for api client library
 # install at compile time so library can be used as soon as needed
-include_recipe 'build-essential'
+build-essential
```
Comment (tas50) [inline recipes/_connect_rundeck_api_client.rb]: This needs to be the resource which is "build_essential" not "build-essential"

Try changing it to

```ruby
build_essential 'install compilation tools for gem install'
```

File: metadata.rb [hunk 2]
```diff
@@ -9,7 +9,6 @@
 depends          'sudo'
 depends          'java'
 depends          'apache2'
-depends          'build-essential'
```
Comment (tas50) [inline metadata.rb]: We still want this for Chef 13 compatibility.

**Comment by tas50** [comment]
You want to keep the depends on build-essential cookbook and still use the resource. A long time ago we added a resource to the build-essential cookbook, but not many people used it. In Chef 14 we shipped that within chef-client. The first step in updating things is to move people over to the resource, but still depend on the cookbook.

**Comment by mrala** [comment]
@tas50 gotcha, thanks for the info.

File: .kitchen.dokken.yml [hunk 3]
```diff
@@ -15,56 +18,56 @@ verifier:
...
 
 platforms:
-- name: debian-7
-  driver:
-    image: debian:7
-    pid_one_command: /sbin/init
-    intermediate_instructions:
-      - RUN /usr/bin/apt-get update
-      - RUN /usr/bin/apt-get install apt-transport-https lsb-release procps net-tools -y
+  - name: debian-7
```
Comment (tas50) [inline .kitchen.dokken.yml]: Debian 7 is EOL so this one can just get deleted

File: recipes/_connect_rundeck_api_client.rb [hunk 4]
```diff
@@ -1,6 +1,10 @@
 # rest-client gem is needed for api client library
 # install at compile time so library can be used as soon as needed
-include_recipe 'build-essential'
+
+build_essential 'install build tools' do
```
Comment (tas50) [inline recipes/_connect_rundeck_api_client.rb]: 👍

File: recipes/_connect_rundeck_api_client.rb [hunk 5]
```diff
@@ -1,6 +1,10 @@
...
+  compile_time true
```
Comment (tas50) [inline recipes/_connect_rundeck_api_client.rb]: You can delete the build-essential attribute in the attributes file now

**Comment by mrala** [comment]
Trying to figure out why unit tests are still failing...

**Comment by damacus** [comment]
Closing this down now, thanks very much for taking the time on this PR! 

@PowerSchill's done some great work that's now in master.

**Pull request closed without merging**